packages/xod-client/src/project/selectors.js

```diff
@@ -149,10 +144,13 @@ const addDeadFlag = R.curry(
 
 // :: State -> StrMap RenderableNode
 export const getRenderableNodes = createMemoizedSelector(
-  [getProject, getCurrentPatchUnsafe, getCurrentPatchNodes, getConnectedPins],
+  [getProject, getCurrentPatch, getCurrentPatchNodes, getConnectedPins],
   [R.T, R.equals, R.equals, R.equals],
-  (project, currentPatch, currentPatchNodes, connectedPins) =>
-    R.map(
+  (project, maybeCurrentPatch, currentPatchNodes, connectedPins) => {
+    if (Maybe.isNothing(maybeCurrentPatch)) return {};
+    const currentPatch = explode(maybeCurrentPatch);
```

Comment:
Uff imperative. Isn’t it better to `foldMaybe`?

Reply:
Yep, it's better. Coming soon.